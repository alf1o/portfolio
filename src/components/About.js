import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  container: {
    marginTop: '1%',
    marginBottom: '50px'
  },
  imgContainer: {
    float: 'left',
    padding: theme.spacing.unit,
    marginRight: '10px',
    textAlign: 'center'
  },
  img: {
    margin: 'auto',
    width: '22vmin',
    height: '22vmin'
  },
  textContainer: {
    margin: '0 3%'
  },
  text: {
    lineHeight: 1.6,
    marginBottom: '15px',
    [theme.breakpoints.down('md')]: {
      fontSize: theme.typography.fontSize + 2
    }
  }
});

/**
  Render an avatar image and some background text.
*/
// TODO: image
function About({ classes, text }) {
  return (
    <Paper
      elevation={0}
      className={classes.container}
    >
      <Paper
        className={classes.imgContainer}
        elevation={0}
      >
        <Avatar
          alt=""
          src=""
          srcSet=""
          sizes=""
          className={classes.img}
        >
          A
        </Avatar>
        <Typography
          variant="caption"
        >
          Alfio Parisi
        </Typography>
        <Typography
          variant="caption"
        >
          London, UK
        </Typography>
      </Paper>
      <Paper
        className={classes.textContainer}
        elevation={0}
      >
        {text.split('\n').map((line, i) => (
          <Typography key={i}
            component="p"
            className={classes.text}
          >
            {line}
          </Typography>
        ))}
      </Paper>
    </Paper>
  );
}

About.propTypes = {
  classes: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired
};

export { About };
export default withStyles(styles)(About);
