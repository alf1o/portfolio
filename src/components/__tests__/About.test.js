import React from 'react';
import { About } from '../About';
import { createShallow } from '@material-ui/core/test-utils';

describe('`About`', () => {
  let props = {};
  let mountedAbout;
  const shallow = createShallow();

  function about() {
    if (!mountedAbout) mountedAbout = shallow(<About {...props} />);
    return mountedAbout;
  }

  beforeEach(() => {
    props = {
      classes: {
        container: {},
        imgContainer: {},
        img: {},
        textContainer: {},
        text: {}
      },
      text: 'Example\n text'
    }
  });

  it('should render as many `p.text` as line breaks in `props.text` + 1', () => {
    const lines = props.text.split('\n').length;
    const ps = about().find({ component: 'p' }).length;
    expect(ps).toBe(lines);
  });

  it('should match snapshot', () => {
    expect(about()).toMatchSnapshot();
  });
});
