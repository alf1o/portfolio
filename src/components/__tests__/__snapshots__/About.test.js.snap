// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`About\` should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About
    classes={
        Object {
            "container": Object {},
            "img": Object {},
            "imgContainer": Object {},
            "text": Object {},
            "textContainer": Object {},
          }
    }
    text="Example
 text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Paper)
          className={Object {}}
          elevation={0}
>
          <WithStyles(Avatar)
                    alt="Alfio's profile picture"
                    className={Object {}}
                    src="profile_pic-min.jpg"
          />
          <WithStyles(Typography)
                    variant="caption"
          >
                    Alfio Parisi
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    variant="caption"
          >
                    London, UK
          </WithStyles(Typography)>
</WithStyles(Paper)>,
        <WithStyles(Paper)
          className={Object {}}
          elevation={0}
>
          <WithStyles(Typography)
                    className={Object {}}
                    component="p"
          >
                    Example
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    className={Object {}}
                    component="p"
          >
                     text
          </WithStyles(Typography)>
</WithStyles(Paper)>,
      ],
      "className": Object {},
      "elevation": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Avatar)
              alt="Alfio's profile picture"
              className={Object {}}
              src="profile_pic-min.jpg"
/>,
            <WithStyles(Typography)
              variant="caption"
>
              Alfio Parisi
</WithStyles(Typography)>,
            <WithStyles(Typography)
              variant="caption"
>
              London, UK
</WithStyles(Typography)>,
          ],
          "className": Object {},
          "elevation": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Alfio's profile picture",
              "className": Object {},
              "src": "profile_pic-min.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Alfio Parisi",
              "variant": "caption",
            },
            "ref": null,
            "rendered": "Alfio Parisi",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "London, UK",
              "variant": "caption",
            },
            "ref": null,
            "rendered": "London, UK",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className={Object {}}
              component="p"
>
              Example
</WithStyles(Typography)>,
            <WithStyles(Typography)
              className={Object {}}
              component="p"
>
               text
</WithStyles(Typography)>,
          ],
          "className": Object {},
          "elevation": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": "Example",
              "className": Object {},
              "component": "p",
            },
            "ref": null,
            "rendered": "Example",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": " text",
              "className": Object {},
              "component": "p",
            },
            "ref": null,
            "rendered": " text",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Paper)
            className={Object {}}
            elevation={0}
>
            <WithStyles(Avatar)
                        alt="Alfio's profile picture"
                        className={Object {}}
                        src="profile_pic-min.jpg"
            />
            <WithStyles(Typography)
                        variant="caption"
            >
                        Alfio Parisi
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        variant="caption"
            >
                        London, UK
            </WithStyles(Typography)>
</WithStyles(Paper)>,
          <WithStyles(Paper)
            className={Object {}}
            elevation={0}
>
            <WithStyles(Typography)
                        className={Object {}}
                        component="p"
            >
                        Example
            </WithStyles(Typography)>
            <WithStyles(Typography)
                        className={Object {}}
                        component="p"
            >
                         text
            </WithStyles(Typography)>
</WithStyles(Paper)>,
        ],
        "className": Object {},
        "elevation": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Avatar)
                alt="Alfio's profile picture"
                className={Object {}}
                src="profile_pic-min.jpg"
/>,
              <WithStyles(Typography)
                variant="caption"
>
                Alfio Parisi
</WithStyles(Typography)>,
              <WithStyles(Typography)
                variant="caption"
>
                London, UK
</WithStyles(Typography)>,
            ],
            "className": Object {},
            "elevation": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Alfio's profile picture",
                "className": Object {},
                "src": "profile_pic-min.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Alfio Parisi",
                "variant": "caption",
              },
              "ref": null,
              "rendered": "Alfio Parisi",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "London, UK",
                "variant": "caption",
              },
              "ref": null,
              "rendered": "London, UK",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className={Object {}}
                component="p"
>
                Example
</WithStyles(Typography)>,
              <WithStyles(Typography)
                className={Object {}}
                component="p"
>
                 text
</WithStyles(Typography)>,
            ],
            "className": Object {},
            "elevation": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": "Example",
                "className": Object {},
                "component": "p",
              },
              "ref": null,
              "rendered": "Example",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": " text",
                "className": Object {},
                "component": "p",
              },
              "ref": null,
              "rendered": " text",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
